programs/gateway/src/instructions/admin.rs

```diff
@@ -50,14 +56,124 @@ pub fn update_tss(ctx: Context<UpdateTss>, tss_address: [u8; 20]) -> Result<()>
     Ok(())
 }
 
-// Updates the PDA authority. Caller is authority stored in PDA.
+// Migrates existing PDA to new format by reallocating account space.
+// This is a one-time migration that can be called after program upgrade.
+pub fn migrate_pda(ctx: Context<MigratePda>) -> Result<()> {
+    let pda_info = &ctx.accounts.pda.to_account_info();
+
+    // Check if already migrated by comparing account size
+    let current_size = pda_info.data_len();
+    let new_size = 8 + std::mem::size_of::<Pda>();
+
+    // If already migrated (size matches new size), do nothing
+    if current_size >= new_size {
+        msg!("PDA already migrated, skipping");
+        return Ok(());
+    }
+
+    // The realloc constraint in the context has already resized the account
```

Comment:
do we need to make sure that pda with more data inside will contain enough funds for rent exempt? maybe that should be documented somewhere, even though it will probably have enough since deposits go to pda anywayrs